### ISSUE TITLE
[production/AQM.v7] DATA directory is not removed at the end of a job. 

### ISSUE THREAD

**Issue by lgannoaa** [ufs-community/ufs-srweather-app#754]
There are a few jobs access files/directory from DATA directory of previous jobs.
For example, the forecast job:
(1) The post job access forecast job output files from the forecast job DATA directory. 
(2) The forecast restart job access forecast job RESTART directory from the previous failed forecast job DATA directory.
(3) The post_stat job however, access forecast job output from COM.

There are two solutions:
(1) Implement a new job to do DATA clean up after DATA directories of a cycle no longer need to be used.
(2) Modify all jobs to copy output to COM and ensure no job access any file from previous job's DATA directory. This is preferred solution because NCO production used KEEPDATA=NO. That is by default the DATA directory used by a job will be deleted when job is completed. 

The bottom line is the AQM production system can not leave DATA without clean up.
## Expected behavior
DATA directory been removed when the job complete and KEEPDATA=NO.

## Current behavior

DATA directory (stmp) not removed. The system stmp get fill up incorrectly.

## Machines affected
This issue is focused on AQM implementation on WCOSS system.

## Steps To Reproduce
Let the job finish and review the DATA directory to see it still not removed. Exam the job log and see it access files from DATA directory of a previous job step. For example, post access files from DATA directory in forecast step.

**Comment by chan-hoo** [comment]
@lgannoaa, it is allowed in the NCO standards. Please check pp.13 (viii Working directories). "Working directories must contain a unique identifier **unless** there is an application need to share the directory across multiple jobs (**e.g. a forecast job writing output that is needed by a post job running in parallel)**. This was confirmed by the EE2 reviewer. We'll just need to remove them once the entire run is completed.

**Comment by lgannoaa** [comment]
@chan-hoo The issue in here is to talk about there is a need to clean up the DATA after job/cycle completed. Currently there is no cleanup job does that. This issue is not intend to say the access of the DATA from previous job step is wrong approach as long as it is approved solution.

**Comment by lgannoaa** [comment]
This issue also highlighted the fact that post is access forecast output from the DATA location from the forecast job. However, the post_stat job run after the post job accessed the forecast output in COM location. This in inconsistency. Since the forecast job will copy all output files to COM, jobs run after forecast step should be able to access fcst output from COM.

**Comment by chan-hoo** [comment]
> This issue also highlighted the fact that post is access forecast output from the DATA location from the forecast job. However, the post_stat job run after the post job accessed the forecast output in COM location. This in inconsistency. Since the forecast job will copy all output files to COM, jobs run after forecast step should be able to access fcst output from COM.

In pp.4 of the NCO standards, "All data that is need for longer than one cycle must be copied to COMOUT". I think we can interpret it as we don't have to use the output files in COMOUT because the run_post and run_fcst jobs are in the same cycle. @MatthewPyle-NOAA, am I correct?

**Comment by chan-hoo** [comment]
@lgannoaa, BTW, the output files of the forecast job are copied to COM once the forecast job is completed. However, the run_post task starts when an output file exists in the working directory of the run_fcst task. Therefore, the run_post task should use the data in the working directory of the run_fcst. Otherwise, the run_post task will start only after run_fcst is finished.

**Comment by lgannoaa** [comment]
> @lgannoaa, BTW, the output files of the forecast job are copied to COM once the forecast job is completed. However, the run_post task starts when an output file exists in the working directory of the run_fcst task. Therefore, the run_post task should use the data in the working directory of the run_fcst. Otherwise, the run_post task will start only after run_fcst is finished.
@JianpingHuang-NOAA instructed me that the ecflow post step run after forecast step is completed. Therefore, all fcst job output will be in COM when post jobs are running.

**Comment by JianpingHuang-NOAA** [comment]
@chan-hoo @lgannoaa   all the run_post tasks can be submitted simultaneously after the run_fcst job is done.

**Comment by chan-hoo** [comment]
@lgannoaa, the UFS SRW App will be used for RRFS implementation soon. RRFS doesn't copy the result files 'phy' and 'dyn' of run_fcst to COM directory. RRFS only stores the output files of run_post in COM. If we change its logic only for AQM, it will be very hard to maintain multiple applications based on the UFS SRW App. I don't think it violates the NCO standards. @MatthewPyle-NOAA, what do you think about this? Is this acceptable?

**Comment by chan-hoo** [comment]
> @lgannoaa, the UFS SRW App will be used for RRFS implementation soon. RRFS doesn't copy the result files 'phy' and 'dyn' of run_fcst to COM directory. RRFS only stores the output files of run_post in COM. If we change its logic only for AQM, it will be very hard to maintain multiple applications based on the UFS SRW App. I don't think it violates the NCO standards. @MatthewPyle-NOAA, what do you think about this? Is this acceptable?

=> I found a solution. I'll add another if-statement for this to ex-script for run_post.

**Comment by MatthewPyle-NOAA** [comment]
@chan-hoo Don't worry about what RRFS will do - what you are doing for production AQMv7 really shouldn't feed back to the main SRW app.   Also as you noted, copying model output to COM is not required (it doesn't make sense for high resolution domains with large model output - wastes disk I/O and disk space).

**Comment by chan-hoo** [comment]
Updated: https://github.com/ufs-community/ufs-srweather-app/blob/7095ddc7a94e7ae4e6b8b734d1da234f69faac41/scripts/exregional_run_post.sh#L176

**Comment by chan-hoo** [comment]
@lgannoaa, the workflow already has an option for removing the working directory (DATA). You can set "KEEPDATA: false" in your config.yaml file: https://github.com/ufs-community/ufs-srweather-app/blob/7095ddc7a94e7ae4e6b8b734d1da234f69faac41/ush/job_preamble.sh#L160

I am testing it now

**Comment by lgannoaa** [comment]
Please ensure:
(1) The KEEPDATA must be coded to allow workflow level overwrite. The ecflow will assign this value.
(2) The application logic for clean up DATA should ensure the target DATA no longer needed for any jobs. There are only some jobs (not all of the jobs) keep DATA for other job to access. this kind of dependency should be tested carefully. Unconditional remove DATA using preamble will result un-recoverable failure.
Thank you for your support.

**Comment by chan-hoo** [comment]
@lgannoaa, 1) I've added KEEPDATA to 'job_preamble' as follows: https://github.com/ufs-community/ufs-srweather-app/blob/4ac769f6278bf150369a0bf5381ce0904fc6b3e0/ush/job_preamble.sh#L127

2) it works well, but I had to change 'job_postamble' as follows: https://github.com/ufs-community/ufs-srweather-app/commit/170b4f518ac664ac5a68dd8a5e4b2275f8d3c88c, I have the shared working directories like `run_fcst` removed when the last 'run_post' task is completed successfully.
